### PR TITLE
Bash autocompletion for task

### LINKED ISSUE #103: Add Bash, ZSH and Windows command line completion
Add bash and zsh autocomplete feature.

### PR DESCRIPTION
**Pull request by bfarayev**
Adds the bash autocomplete script into the repo (#103)

How to test it in Mac OS:

First make sure `bash-completion` is installed in your machine:

```shell
$ brew install bash-completion
```

It'll create `/usr/local/etc/bash_completion.d/` folder and add autocomplete files for your existing CLI tools.

For it to work, add this to your `~/.bashrc`:
```shell
[ -f /usr/local/etc/bash_completion ] && . /usr/local/etc/bash_completion
```

Now we're ready to add `task` autocomplete script:

Copy the bash autocomplete script and put it to `/usr/local/etc/bash_completion.d/task`. Start a new shell and it should work!

It's open to the feedback. I think we should add more documentation about autocompletion setup, as some users may not even know it exists for task.

### CODE REVIEW

**Comment by andreynering** [comment]
Hi @bfarayev, thanks for the PR!

Help with completion is very welcome since knowledge on how completion how is kinda limited.

/cc @sawadashota and @GuillaumeAmat (previous zsh completion contributors): do you guys mind taking at look at this PR?

> It's open to the feedback. I think we should add more documentation about autocompletion setup, as some users may not even know it exists for task.

Indeed. Also, I'm not sure if possible, but it'd be nice if we could ship completion with Homebrew, Snapcraft, etc.

**Comment by GuillaumeAmat** [comment]
Hi @andreynering, @bfarayev,

The CI is red and I have an issue with the completion script. It keeps the colon after the completed command (eg: `$ task upgrade:` instead of just `$ task upgrade`).

Apart from that issue, it seems to work great!

**Comment by andreynering** [comment]
@GuillaumeAmat Thanks for reviewing!

**Comment by bfarayev** [comment]
Thanks @GuillaumeAmat! I'll try to fix the colon issue but so far I couldn't reproduce that. will have closer look.

also @andreynering do you have any idea why pipeline would fail for go1.12 build? We're adding a file which shouldn't affect go builds at all.

**Comment by andreynering** [comment]
> do you have any idea why pipeline would fail for go1.12 build? We're adding a file which shouldn't affect go builds at all.

I think Travis is just intermittently failing. I just noticed that on other PR as well. A push should retrigger the build and make it pass.

**Comment by bfarayev** [comment]
I've looked at it but unfortunately I couldn't reproduce the issue. I'm suspecting that colon issue might there for 2 reasons:
- our bash versions are different but it's unlikely imho. I'm using 
```
$ bash --version
GNU bash, version 5.0.11(1)-release (x86_64-apple-darwin18.6.0)
```
- `$ task upgrade` in this example has a subtask and tab autocompletion wants to continue. I'm using this project itself to test the script and here's what happens:
```
bash-5.0$ task 
ci                       dl-deps                  generate-install-script  lint                     test-release             
clean                    docs:install             go-get                   packages                 update-deps              
default                  docs:serve               install                  test                     vendor 
```
I can see that we have `Taskfile.yml` has include line which also has `install` and `serve` in it.
```
includes:
  docs: ./docs
```
so I want autocomplete to work for `docs:install` and `docs:serve` and I works as expected for me. Other single commands like `task ci`, `task clean` etc also work as expected.

in summary, when this is what it looks like for me:

`task doc<Tab>` -> `docs:`
`task docs:<Tab>` -> `docs:install docs:serve`
`task docs:ins<Tab` -> `task docs:install`

I'll have a closer look again. Will keep you guys updated.

**Comment by andreynering** [comment]
Hi @bfarayev, sorry for the wait!

I gonna merge this now. If people find bugs, they can report or send pull requests.

Thanks a lot for contributing! 😉
